LSPS2/README.md

```diff
@@ -0,0 +1,736 @@
+>   - PTLCs would allow a payer to buy a private key from a payee
+>     (client in this context); normally the per-hop blinding factors
+>     would hide the sold private key from forwarding nodes, but due to
+>     the LSP having to know the exact payment point and the blinding
+>     factor towards the client / payee, it would also learn the sold
+>     private key.
+
+### 1. API Information
+
+`lsps2.getinfo` is the entrypoint for each client using the API.
+It indicates supported versions of this protocol, as well as any limits the
+LSP imposes, and parameters for payment.
+
+The client MUST request `lsps2.getinfo` to read:
+
+- the supported `versions` of the LSP.
+- the `opening_fee` of the LSP and its related parameters.
+
+The client MUST then determine protocol compatibility (if it supports a
+`version` that the LSP also supports), and also determine if the fee
+imposed by the LSP is acceptable.
+
+`lsps2.getinfo` takes no parameters `{}`, and has no errors defined.
+
+Example `lsps2.getinfo` result:
+
+```JSON
+{
+    "versions": [1],
+    "opening_fee_params_menu": [
+        {
+            "base": "546000",
+            "proportional": 1200,
+            "valid_until": "2023-02-23T08:47:30.511Z",
+            "max_idle_time": 1008,
+            "max_client_to_self_delay": 2016,
+            "promise": "abcdefghijklmnopqrstuvwxyz"
+        },
+        {
+            "base": "1092000",
+            "proportional": 2400,
+            "valid_until": "2023-02-27T21:23:57.984Z",
+            "max_idle_time": 1008,
+            "max_client_to_self_delay": 2016,
+            "promise": "abcdefghijklmnopqrstuvwxyz"
+        }
+    ],
+    "min_payment_size": "1000",
+    "max_payment_size": "1000000",
+}
+```
+
+`versions` is the set of LSPS2 versions the LSP supports.
+When the client later contacts the LSP, it indicates a single specific
+version, which MUST be one of those indicated in this set.
+
+The `opening_fee_params_menu` is an array of `opening_fee_params`
+objects.
+The LSP MUST return a menu with at least one entry.
+
+An `opening_fee_params` object describes how much the LSP will charge
+for a channel open, and until when the described feerates will be
+considered valid.
+An `opening_fee_params` object MUST have all of the following fields,
+and MUST NOT have any additional fields:
+
+* `base` is the number of millisatoshis that the client has to pay for the
+  channel open, regardless of how large the payment is.
+* `proportional` is a parts-per-million number that describes how many
+  millisatoshis to charge for every 1 million millisatoshis of payment
+  size for the first payment.
+* `valid_until` is a datetime (as an ISO8601 string) up to which this specific
+  `opening_fee_params` is valid, and also serves as the timeout for the JIT
+  Channel flow, if this particular object is selected.
+* `max_idle_time` is a maximum number of blocks where the LSP promises to
```

Comment:
This is a minimum I think?

Reply:
Saying "minimum" here would be confusing with the `max` in the name, hmmm.  So maybe reword:

```Markdown
* `max_idle_time` is a number of blocks that the LSP will wait for until
  it closes the channel for inactivity; the LSP MAY close the channel
  if at least that number of blocks has passed since activity occured
  on the channel.
```

This avoids using "minimum" or "maximum" in the definition.

Reply:
That looks good to me. 👍

Reply:
Done.